### PR TITLE
feat: support local storage

### LINKED ISSUE #344: 图片、文件、视频能否支持本地存储？
### Is your feature request related to a problem?

需求点：
- 现在图片或文件都放数据库里加载出来有点慢，每次刷新页面都会重新从数据库拉取一次文件，无法像静态资源那样缓存在浏览器
- 私人照片不想放图床
- 视频有的比较大，也不方便用网络存储

### Describe the solution you'd like

- 支持本地存储
- 支持存视频文件，视频播放支持断点续传
- 本地图片自动生成缩略图，小图使用缩略图，点击再加载原图

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Zeng1998**
Close #344 #1163 #959 .

### CODE REVIEW

**Comment by boojack** [comment]
Failed to upload

image

File: server/memo.go [hunk 1]
```diff
@@ -202,7 +202,6 @@ func (s *Server) registerMemoRoutes(g *echo.Group) {
 		if offset, err := strconv.Atoi(c.QueryParam("offset")); err == nil {
 			memoFind.Offset = &offset
 		}
-
```
Comment (boojack) [inline server/memo.go:205]: Looks like an unrelated change

**Review by boojack — APPROVED** [review]
LGTM

**Comment by Zeng1998** [comment]
fixed it.
